### PR TITLE
Extract ONNX_API macro to a common header and export OpSchemaRegistry

### PR DESCRIPTION
**Pull request by bddppq**

### CODE REVIEW

File: onnx/api_macros.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#ifndef ONNX_API_H
```
Comment (houseroad) [inline onnx/api_macros.h]: I think better to use `ONNX_API_MACROS_H`
in case some day we have onnx/api.h

Comment (bddppq) [reply onnx/api_macros.h]: oops that's because I initially named this file as onnx/api.h

File: cmake/Utils.cmake [hunk 2]
```diff
@@ -27,4 +27,11 @@ function(add_onnx_global_defines target)
     target_compile_definitions(${target} PUBLIC "ONNX_USE_LITE_PROTO=1")
   endif()
 
+  if (BUILD_SHARED_LIBS)
+    target_compile_definitions(${target} PRIVATE "ONNX_BUILD_SHARED_LIBS=1")
+  endif()
+
+  if (ONNX_BUILD_MAIN_LIB)
+    target_compile_definitions(${target} PRIVATE "ONNX_BUILD_MAIN_LIB=1")
```
Comment (yinghai) [inline cmake/Utils.cmake]: What's the purpose of these build variants?

Comment (zrphercule) [reply cmake/Utils.cmake]: @yinghai See the the followed api_macros.h, they are used there. I think it is used to deal with some ms dll stuff.

**Pull request closed without merging**